bsmetadata/train.py

```diff
@@ -149,12 +173,16 @@ def main(args: CFG) -> None:
     os.makedirs(args.out_dir, exist_ok=True)
 
     # get dataloaders
-    tokenizer = AutoTokenizer.from_pretrained(args.model_name)
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
     tokenizer.pad_token = tokenizer.eos_token
     train_dataloader, eval_dataloaders = get_dataloaders(tokenizer, args.data_config)
 
     # get model
-    model = AutoModelForCausalLM.from_pretrained(args.model_name)
+    if not resumed_state:
+        model = AutoModelForCausalLM.from_pretrained(model_name)
+    else:
+        print("Loading model from checkpoint")
+        model = AutoModelForCausalLM.from_pretrained(resumed_state, from_pt=True)
```

Comment:
Does `from_pretrained` accept a state_dict (`resumed_state`)?
I don't think this is correct, but I did not try it.

I would `        model = AutoModelForCausalLM.from_pretrained(model_name)`
then `model.load_state_dict` if `resumed_state`

And put all of the if-then-load together in a single `if`

Reply:
Thank you. Yeah it does make more sense to put scheduler loading there, together with scheduler creation.

LGTM.